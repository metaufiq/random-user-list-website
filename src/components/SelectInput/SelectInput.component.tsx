import { Fragment } from "react";
import { Props } from "./SelectInput.component.types";

const SelectInput = ({options, label, htmlFor, id, onChange, value}: Props)=>{
  return(
    <Fragment>
      {label && 
        <label 
          htmlFor={htmlFor} 
          className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400"
        >
        {label}
        </label>}
      <select 
        id={id} 
        className="bg-white border border-gray-300 text-gray-900 text-sm rounded-sm focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5"
        value={value}
        onChange={onChange}>
        {options.map(option=><option value={option.value}>{option.label}</option>)}
      </select>
    </Fragment>
  )
}

export default SelectInput;