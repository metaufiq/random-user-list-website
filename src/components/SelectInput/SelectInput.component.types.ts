import { User } from "../../index.types"

export type Option = {
  value: any,
  label: string
}

export type Props = {
  options: Option[]
}