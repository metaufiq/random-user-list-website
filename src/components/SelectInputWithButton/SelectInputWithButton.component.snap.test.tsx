import { assertSnapshots } from "../../../test";
import SelectInput from ".";
import { Props } from "./SelectInputWithButton.component.types";

describe('SelectInputWithButton Snapshot', () => {
  const mockProps:Props = {
    onChange:()=>{},
    label: 'label',
    resetButtonLabel: 'reset button label',
    options: [{
      label: 'label',
      value: 'value'
    }],
    onReset: ()=>{}
  };

  const mockPropsWithOptionalProps: Props = {
    ...mockProps,
    htmlFor: 'htmlFor',
    id: 'id',
    value: 'value'
  }

  const configs = [
    {
      props: mockProps,
      desc: 'should render normal SelectInputWithButton'
    },
    {
      props: mockPropsWithOptionalProps,
      desc: 'should render normal SelectInputWithButton with optional props'
    }
  ];

  assertSnapshots(SelectInput, configs);
});
