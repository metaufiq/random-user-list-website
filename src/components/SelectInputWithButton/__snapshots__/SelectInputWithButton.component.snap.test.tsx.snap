// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInputWithButton Snapshot should render normal SelectInputWithButton 1`] = `
Array [
  <label
    className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400"
  >
    label
  </label>,
  <div
    className="flex flex-row"
  >
    <div
      className="mr-2 flex-auto"
    >
      <select
        className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        onChange={[Function]}
      >
        <option
          value="value"
        >
          label
        </option>
      </select>
    </div>
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
      onClick={[Function]}
      type="button"
    >
      reset button label
    </button>
  </div>,
]
`;

exports[`SelectInputWithButton Snapshot should render normal SelectInputWithButton with optional props 1`] = `
Array [
  <label
    className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400"
    htmlFor="htmlFor"
  >
    label
  </label>,
  <div
    className="flex flex-row"
  >
    <div
      className="mr-2 flex-auto"
    >
      <select
        className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        id="id"
        onChange={[Function]}
        value="value"
      >
        <option
          value="value"
        >
          label
        </option>
      </select>
    </div>
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
      onClick={[Function]}
      type="button"
    >
      reset button label
    </button>
  </div>,
]
`;
